### PR TITLE
move manifests stuff to its own pkg and add OCI mime types

### PR DESCRIPTION
**Pull request by runcom**
Extracted from #102 - pretty self contained change, I'll rebase #102 once this is merged

@mtrmac PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@mtrmac this should be pretty straight forward to merge if you agree on such a structure (which you did IIRC in #102)

File: manifest/manifest.go [hunk 1]
```diff
@@ -34,9 +48,10 @@ func GuessManifestMIMEType(manifest []byte) string {
 	}
 
 	switch meta.MediaType {
-	case DockerV2Schema2MIMEType, DockerV2ListMIMEType: // A recognized type.
+	case DockerV2Schema2MIMEType, DockerV2ListMIMEType, OCIV1DescriptorMIMEType, OCIV1ImageManifestMIMEType, OCIV1ImageManifestListMIMEType, OCIV1ImageSerializationRootfsTarGzipMIMEType, OCIV1ImageSerializationConfigMIMEType, OCIV1ImageSerializationCombinedMIMEType: // A recognized type.
 		return meta.MediaType
 	}
+	// TODO(mitr, runcom): flacky? just error out?
```
Comment (mtrmac) [inline manifest/manifest.go]: Reiterating https://github.com/projectatomic/skopeo/pull/102#discussion-diff-66969257 .  LGTM after dropping this comment.

Comment (mtrmac) [inline manifest/manifest.go]: It might even be worth it to explicitly add a comment describing why this is necessary.

Comment (runcom) [inline manifest/manifest.go]: droppend and added a comment from #102 discussion

**Comment by mtrmac** [comment]
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

somewhere during DNF download; is there a way to refresh the test?

**Pull request merged**